pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -339,33 +339,40 @@ func (ku *KubeUtil) resetCredentials() {
 	ku.rawConnectionInfo = make(map[string]string)
 }
 
-// QueryKubelet allows to query the KubeUtil registered kubelet API on the parameter path
-// path commonly used are /healthz, /pods, /metrics
-// return the content of the response, the response HTTP status code and an error in case of
-func (ku *KubeUtil) QueryKubelet(path string) ([]byte, int, error) {
+func (ku *KubeUtil) doKubeletRequest(path string) (*http.Response, error) {
 	var err error
 
 	req := &http.Request{}
 	req.Header = *ku.kubeletApiRequestHeaders
-	req.URL, err = url.Parse(fmt.Sprintf("%s%s", ku.kubeletApiEndpoint, path))
+	req.URL, err = url.Parse(ku.kubeletApiEndpoint + path)
 	if err != nil {
 		log.Debugf("Fail to create the kubelet request: %s", err)
-		return nil, 0, err
+		return nil, err
 	}
 
 	response, err := ku.kubeletApiClient.Do(req)
 	if err != nil {
 		log.Debugf("Cannot request %s: %s", req.URL.String(), err)
+		return nil, err
+	}
+	return response, nil
+}
+
+// QueryKubelet allows to query the KubeUtil registered kubelet API on the parameter path
+// path commonly used are /healthz, /pods, /metrics
+// return the body of the response, the response HTTP status code and an error in case of
+func (ku *KubeUtil) QueryKubelet(path string) ([]byte, int, error) {
+	response, err := ku.doKubeletRequest(path)
+	if err != nil {
 		return nil, 0, err
 	}
 	defer response.Body.Close()
-
 	b, err := ioutil.ReadAll(response.Body)
```

Comment:
This could be replaced by something like [this](https://github.com/DataDog/pupernetes/blob/master/pkg/run/getter.go#L31-L33) if more effective.
@xvello what do you think ?

Reply:
I'm not sure of this idea, because using a json decoder should only work with the `/pods` or other json based endpoints...

Reply:
No problem as long as we handle the http code before, to avoid "json parsing error" because we have a 403 (cf our previous apiserver library). 
We use it to query non-json endpoints though: `/healthz` and `/metrics` (to get the kubelet version from the prometheus metric, not the actually container metrics. We'd need two different methods (json & text)